tests/python/relay/test_qnn_conv2d.py

```diff
@@ -0,0 +1,626 @@
+            kernel_layout=kernel_layout)
+
+    mod = relay.Function(relay.analysis.free_vars(func), func)
+    mod = relay.Module.from_expr(mod)
+    mod = relay.qnn.transform.CanonicalizeOps()(mod)
+    return mod
+
+def get_funcs(data_shape,
+              data_dtype,
+              kernel_shape,
+              kernel_dtype,
+              input_zero_point,
+              kernel_zero_point,
+              kernel_size,
+              padding,
+              strides,
+              dilation,
+              data_layout,
+              kernel_layout,
+              out_dtype):
+    data = relay.var("data", shape=data_shape,
+            dtype=data_dtype)
+    kernel = relay.var("kernel", shape=kernel_shape,
+            dtype=kernel_dtype)
+    ref_func = get_ref_func(data,
+                            kernel,
+                            input_zero_point,
+                            kernel_zero_point,
+                            kernel_size,
+                            padding,
+                            strides,
+                            dilation,
+                            data_layout,
+                            kernel_layout,
+                            out_dtype)
+    ref_func = run_infer_type(ref_func)
+    qnn_func = get_qnn_func(data,
+                            kernel,
+                            input_zero_point,
+                            kernel_zero_point,
+                            kernel_size,
+                            padding,
+                            strides,
+                            dilation,
+                            data_layout,
+                            kernel_layout,
+                            out_dtype)
+    return (ref_func, qnn_func)
+
+def verify(ref_func, qnn_func, data_shape, data_dtype, kernel_shape,
+        kernel_dtype):
+    def get_inputs(data_shape, data_dtype, kernel_shape, kernel_dtype):
+        # Keeping inputs multiple of 4 because of a bug in Average Pool2d
+        # https://discuss.tvm.ai/t/pool2d-gives-bad-output-for-integer-inputs/3377
+        low = -128
+        high = 127
+        if data_dtype == "uint8":
+            low = 0
+            high = 255
+        golden_data = np.random.random_integers(low=low, high=high,
+                size=data_shape).astype(data_dtype)
+        low = -128
+        high = 127
+        if kernel_dtype == "uint8":
+            low = 0
+            high = 255
+        golden_weight = np.random.random_integers(low=low, high=high,
+                size=kernel_shape).astype(kernel_dtype)
+        return (golden_data, golden_weight)
+
+
+    def get_output(func, golden_inputs):
+        with relay.build_config(opt_level=2):
+            golden_data, golden_weight = golden_inputs
+            params = {'kernel': golden_weight}
+            graph, lib, params = relay.build(func, "llvm", params=params)
+            mod = graph_runtime.create(graph, lib, ctx=tvm.cpu(0))
+            mod.set_input("data", golden_data)
```

Comment:
nit: move data to `params` dict as well and remove this and the next line

Reply:
Will remove the comment in next line. Thanks for pointing that out.

I think I prefer to keep the data as a var. I have some tests that check for fold constant output. And this is also what happens in reality. I think this should be ok.